src/package/packageBundleVersion.ts

```diff
@@ -0,0 +1,364 @@
+        (error: SfError) => {
+          if (error.name === 'PollingClientTimeout') {
+            const modifiedError = new SfError(error.message);
+            modifiedError.setData({ VersionCreateRequestId: createResult.Id });
+            modifiedError.message += ` Run 'sf package bundle version create report -i ${createResult.Id}' to check the status.`;
+            throw applyErrorAction(massageErrorMessage(modifiedError));
+          }
+          throw applyErrorAction(massageErrorMessage(error));
+        }
+      );
+    }
+
+    return createResult;
+  }
+
+  public static async pollCreateStatus(
+    createPackageVersionRequestId: string,
+    connection: Connection,
+    project: SfProject,
+    polling: { frequency: Duration; timeout: Duration }
+  ): Promise<BundleSObjects.PackageBundleVersionCreateRequestResult> {
+    if (polling.timeout?.milliseconds <= 0) {
+      return PackageBundleVersionCreate.getCreateStatus(createPackageVersionRequestId, connection);
+    }
+    let remainingWaitTime: Duration = polling.timeout;
+    const pollingClient = await PollingClient.create({
+      poll: async (): Promise<StatusResult> => {
+        const report = await PackageBundleVersionCreate.getCreateStatus(createPackageVersionRequestId, connection);
+        switch (report.RequestStatus) {
+          case BundleSObjects.PkgBundleVersionCreateReqStatus.queued:
+            await Lifecycle.getInstance().emit(PackageVersionEvents.create.enqueued, { ...report, remainingWaitTime });
+            remainingWaitTime = Duration.seconds(remainingWaitTime.seconds - polling.frequency.seconds);
+            return {
+              completed: false,
+              payload: report,
+            };
+          case BundleSObjects.PkgBundleVersionCreateReqStatus.success: {
+            await Lifecycle.getInstance().emit(PackageVersionEvents.create.success, report);
+            return { completed: true, payload: report };
+          }
+          case BundleSObjects.PkgBundleVersionCreateReqStatus.error:
+            await Lifecycle.getInstance().emit(PackageVersionEvents.create.error, report);
+            return { completed: true, payload: report };
+        }
+      },
+      frequency: polling.frequency,
+      timeout: polling.timeout,
+    });
+
+    try {
+      return await pollingClient.subscribe<BundleSObjects.PackageBundleVersionCreateRequestResult>();
+    } catch (err) {
+      const report = await PackageBundleVersionCreate.getCreateStatus(createPackageVersionRequestId, connection);
+      await Lifecycle.getInstance().emit(PackageVersionEvents.create['timed-out'], report);
+      if (err instanceof Error) {
+        throw applyErrorAction(err);
+      }
+      throw err;
+    }
+  }
+
+  public static async report(connection: Connection, id: string): Promise<BundleSObjects.BundleVersion | null> {
+    const query =
+      'SELECT Id, PackageBundle.Id, PackageBundle.BundleName, VersionName, MajorVersion, MinorVersion, IsReleased, ' +
```

Comment:
can we use a template string \` xxx \` to avoid multi-line string concats with '+'